quickwit/quickwit-config/src/metastore_config.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::num::NonZeroUsize;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum MetastoreBackend {
+    File,
+    #[serde(alias = "pg", alias = "postgres")]
+    PostgreSQL,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum MetastoreConfig {
+    File(FileMetastoreConfig),
+    #[serde(alias = "pg", alias = "postgres")]
+    PostgreSQL(PostgresMetastoreConfig),
+}
+
+impl MetastoreConfig {
+    pub fn backend(&self) -> MetastoreBackend {
+        match self {
+            Self::File(_) => MetastoreBackend::File,
+            Self::PostgreSQL(_) => MetastoreBackend::PostgreSQL,
+        }
+    }
+
+    pub fn as_file(&self) -> Option<&FileMetastoreConfig> {
+        match self {
+            Self::File(file_metastore_config) => Some(file_metastore_config),
+            _ => None,
+        }
+    }
+
+    pub fn as_postgres(&self) -> Option<&PostgresMetastoreConfig> {
+        match self {
+            Self::PostgreSQL(postgres_metastore_config) => Some(postgres_metastore_config),
+            _ => None,
+        }
+    }
+
+    pub fn redact(&mut self) {
+        // TODO: Implement this method when we end up storing secrets in the
```

Comment:
this is the case for postgresql

Reply:
I have not added the database password in the config in this PR.